feature_engine/prediction/target_mean_classifier.py

```diff
@@ -0,0 +1,192 @@
+            )
+
+        # if target has values other than 0 and 1, we need to remap the values,
+        # to be able to compute meaningful averages.
+        if any(x for x in y.unique() if x not in [0, 1]):
+            y = np.where(y == y.unique()[0], 0, 1)
+
+        super().fit(X, y)
+
+        return self
+
+    def predict_proba(self, X: pd.DataFrame) -> pd.Series:
+        """
+        Proxy for probability estimates based of the average of the target mean value
+        across variables.
+
+        The returned estimates for all classes are ordered by the label of classes.
+
+        Parameters
+        ----------
+        X : pandas dataframe of shape = [n_samples, n_features]
+            The input samples.
+
+        Return
+        -------
+        T: array-like of shape (n_samples, n_classes)
+            Returns the probability of the sample for each class in the model, where
+            classes are ordered as they are in self.classes_.
+        """
+        prob = self._predict(X)
+        if not isinstance(prob, np.ndarray) or prob.shape[1] != 2:
+            raise ValueError(
+                "'prob' is not a 2-dimension numpy array. The probability "
+                "predictions must be a 2-dimension numpy array with 2 columns."
+            )
+        # the second column is the prob of class 1.
+        return np.vstack([1 - prob, prob]).T
+
+    def predict_log_proba(self, X: pd.DataFrame) -> pd.Series:
+        """
+        Predict logarithm of probability estimates.
+
+        The returned estimates for all classes are ordered by the label of classes.
+
+        Parameters
+        ----------
+        X : pandas dataframe of shape = [n_samples, n_features]
+            The input samples.
+
+        Return
+        -------
+        T: array-like of shape (n_samples, n_classes)
+            Returns the log-probability of the sample for each class in the model,
+            where classes are ordered as they are in self.classes_.
+
+        """
+        log_prob = np.log(self.predict_proba(X))
+        if not isinstance(log_prob, np.ndarray) or log_prob.shape[1] != 2:
+            raise ValueError(
+                "'log_prob' is not a 2-dimension numpy array. The log probability "
+                "predictions must be a 2-dimension numpy array with 2 columns."
+            )
+        return log_prob
+
+    def predict(self, X: pd.DataFrame) -> pd.Series:
+        """
+        Predict class labels for samples in X.
+
+        Parameters
+        ----------
+        X : pandas dataframe of shape = [n_samples, n_features]
+            The input series which must have the same name as one of the features in the
+            dataframe that was used to fit the predictor.
+
+        Return
+        -------
+        y_pred: ndarray of shape (n_samples,)
+            Vector containing the class labels for each sample.
+        """
+        y_pred = np.where(self.predict_proba(X) > 0.5, 1, 0)
```

Comment:
is the output of predict_proba not a 2 column array? which probability should be bigger than 0.5?